engine/executor/dm/worker.go

```diff
@@ -302,6 +308,7 @@ func (w *dmWorker) setStage(stage metadata.TaskStage) {
 	w.mu.Lock()
 	defer w.mu.Unlock()
 	w.stage = stage
+	w.stageGauge.Set(float64(stage))
```

Comment:
maybe we can use the metrics factory `w.MetricFactory()`

Reply:
used `w.MetricFactory()`